src/main/java/flaxbeard/steamcraft/block/BlockPipe.java

```diff
@@ -334,10 +334,13 @@ public void registerBlockIcons(IIconRegister ir) {
     @SubscribeEvent
     public void onBlockHighlight(DrawBlockHighlightEvent event) {
         Item equipped = event.player.getCurrentEquippedItem() != null ? event.player.getCurrentEquippedItem().getItem() : null;
-        if ((event.target.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK) && (event.player.worldObj.getBlock(event.target.blockX, event.target.blockY, event.target.blockZ) instanceof BlockPipe)
-                && (event.player.getCurrentEquippedItem() != null) && ((event.player.getCurrentEquippedItem().getItem() instanceof IPipeWrench && ((IPipeWrench) equipped).canWrench(event.player, event.target.blockX, event.target.blockY, event.target.blockZ)))) {
-            RayTracer.retraceBlock(event.player.worldObj, event.player, event.target.blockX, event.target.blockY, event.target.blockZ);
-        }
+       if(equipped!=null){
```

Comment:
What I had in my mind initially is to do simple changes in the code, with the guidance of the SonarQ. As I get to know the code better, I can then tackle changing  the structure of the methods as there is always the danger that the project owner -without much thinking about your efforts - will just  refuse big changes in the beginning.